pkg/appconsts/versioned_consts.go

```diff
@@ -79,18 +80,21 @@ func GetTimeoutCommit(v uint64) time.Duration {
 }
 
 // UpgradeHeightDelay returns the delay in blocks after a quorum has been reached that the chain should upgrade to the new version.
-func UpgradeHeightDelay(v uint64) int64 {
+func UpgradeHeightDelay(chainID string, v uint64) int64 {
 	if OverrideUpgradeHeightDelayStr != "" {
 		parsedValue, err := strconv.ParseInt(OverrideUpgradeHeightDelayStr, 10, 64)
 		if err != nil {
 			panic("Invalid OverrideUpgradeHeightDelayStr value")
 		}
 		return parsedValue
 	}
-	switch v {
-	case v1.Version:
+	switch {
+	case v == v1.Version:
 		return v1.UpgradeHeightDelay
-	case v2.Version:
+	// ONLY ON ARABICA: don't return the v2 value even when the app version is
+	// v2 on arabica. This is due to a bug that was shipped on arabica, where
+	// the next version was used.
+	case v == v2.Version && !strings.Contains(chainID, "arabica"):
```

Comment:
Why not specify it exactly as arabica-11?

Reply:
I would prefer for legibility that you kept the switch v and inside the case v2.Version you have an if statement for arabica-11